Scripts/aws_stack_provision.ps1

```diff
@@ -21,10 +21,10 @@ if ($EXISTING_INSTANCE_COUNT -eq 1 ){
      $INSTANCE_ID = ((Get-EC2Instance -Filter @{ Name="tag:LansaVersion"; Values=$lansa_version }).Instances).InstanceId
      Start-Sleep -Seconds 60
      $NEW_STATUS = (((Start-EC2Instance -InstanceID $INSTANCE_ID).CurrentState).Name).Value
```

Comment:
You need a loop here until its running. And remove the Start-Sleep.
There is a script in the VM in in c:\lansa\scripts\dot-Create-EC2Instance.ps1. Its about creating the instance so that part is not relevant as you are using a template elsewhere in this script. But there is a loop thats attempting to get the password. Only when this is successful is the Windows instance truly ready.

You can't use that script directly of course as you are no on the VM. Copy the code into a Function in your script and adapt it as necessary.

And you need the code for getting the password so you may create/update the AWS Secret. I suggest you ALWAYS update it, even if the instance already exists. If the instance is created for more than 3 months, quite likely, then the password MUST be changed. This is problematic because the password is no longer obtainable via Get-EC2PasswordData, so the excepted behaviour is to terminate the instance and create a new VM.